### PR TITLE
Add support for EXTERNAL_DEPENDENCIES=1

### PR DESCRIPTION
**Pull request by JasonGross**
Based on the scheme in bedrock2

### CODE REVIEW

**Comment by cpitclaudel** [comment]
Is there a plan to move bedrock2 to dune?  It supports composition in subfolders / submodules out of the box with no additional configuration

**Comment by JasonGross** [comment]
> Is there a plan to move bedrock2 to dune? It supports composition in subfolders / submodules out of the box with no additional configuration

That is a question for @samuelgruetter or @andres-erbsen I think.  (I've never set up a dune build)

**Comment by JasonGross** [comment]
Any objections to me merging this so we can make progress on https://github.com/coq/opam-coq-archive/pull/2137#discussion_r835275030?

File: Makefile [hunk 1]
```diff
@@ -37,37 +34,81 @@ clean:: Makefile.coq.lib Makefile.coq
...
+EXTERNAL_DEPENDENCIES?=
+EXTERNAL_COQUTIL?=
+EXTERNAL_BEDROCK2?=
+
 COQUTIL_FOLDER := bedrock2/deps/coqutil
 BEDROCK2_FOLDER := bedrock2/bedrock2
 
+# Note: make does not interpret "\n", and this is intended
+DEPFLAGS_COQUTIL_NL=-Q $(COQUTIL_FOLDER)/src/coqutil coqutil\n
+DEPFLAGS_BEDROCK2_NL=-Q $(BEDROCK2_FOLDER)/src/bedrock2 bedrock2
```
Comment (samuelgruetter) [inline Makefile:46]: This line should probably end with `\n` too, as the variable name suggests?

File: Makefile [hunk 2]
```diff
@@ -37,37 +34,81 @@ clean:: Makefile.coq.lib Makefile.coq
...
+	+$(MAKE) -f Makefile.coq $@
 
-_CoqProject: Makefile
-	@echo "-R src/Rupicola Rupicola" > $@
-	@echo "-Q bedrock2/bedrock2/src/bedrock2 bedrock2" >> $@
-	@echo "-Q bedrock2/deps/coqutil/src/coqutil coqutil" >> $@
-	@echo "-arg -w" >> $@
-	@echo "-arg -unexpected-implicit-declaration,-deprecated-ident-entry" >> $@
+_CoqProject:
+	@printf -- '$(ALLDEPFLAGS_NL)' > _CoqProject
```
Comment (samuelgruetter) [inline Makefile:108]: This repo has a checked-in _CoqProject (whereas bedrock2 doesn't check it in), did you test if you still get the same when you regenerate it?

Comment (JasonGross) [reply Makefile:108]: Maybe we should just remove the checked-in _CoqProject?

File: Makefile (hunk 1 above)
Comment (JasonGross) [reply Makefile:46]:
```suggestion
DEPFLAGS_BEDROCK2_NL=-Q $(BEDROCK2_FOLDER)/src/bedrock2 bedrock2\n
```

File: Makefile (hunk 2 above)
Comment (JasonGross) [reply Makefile:108]: The diff is
```diff
diff --git a/_CoqProject b/_CoqProject
index a939e70..67def96 100644
--- a/_CoqProject
+++ b/_CoqProject
@@ -1,5 +1,5 @@
 -R src/Rupicola Rupicola
--Q bedrock2/bedrock2/src/bedrock2 bedrock2
--Q bedrock2/deps/coqutil/src/coqutil coqutil
 -arg -w
 -arg -unexpected-implicit-declaration,-deprecated-ident-entry
+ -Q bedrock2/deps/coqutil/src/coqutil coqutil
+ -Q bedrock2/bedrock2/src/bedrock2 bedrock2
```
btw

Comment (JasonGross) [reply Makefile:108]: _CoqProject removed

**Comment by cpitclaudel** [comment]
Thanks for working on this.

**Pull request merged**